build-tools/packages/build-tools/src/fluidBuild/fluidPackageCheck.ts

```diff
@@ -488,22 +508,68 @@ export class FluidPackageCheck {
-            const lintChildren = hasPrettier ? ["prettier", "eslint"] : ["eslint"];
-            if (this.checkChildrenScripts(pkg, "lint", lintChildren, false, fix)) {
-                fixed = true;
+            // TODO: add prettier check comment once prettier is enforced globally, hasPrettier commented out to discard build warnings from "lint" & "lint:fix" scripts
+            // const hasPrettier = pkg.getScript("prettier");
+
+            const lintScript = pkg.getScript("lint");
+            const lintFixScript = pkg.getScript("lint:fix");
+
+            /* Regular Expression
+                lintRegex is a regex to extract lint-package name from lintScript & lintFixScript
+
+                if lintScript:
+                    -> lintMatch = ["npm run foo", "npm run bar"]
+                    -> lintChildren = ["foo", "bar"]
+                else:
+                    -> lintFixMatch = ["npm run foo:fix", "npm run bar:fix"]
+                    -> lintFixChildren = ["foo", "bar"]
+            */
+            const lintRegex = /npm run (\w+)(?::fix)?/g;
+
+            // Check "lint"
+            if (lintScript) {
+                const lintMatch = lintScript?.match(lintRegex);
+                const lintChildren = lintMatch?.map((e) => e.replace(/npm run |:fix/g, ""));
+
+                if (lintChildren !== undefined) {
+                    const hasEslint = lintChildren.some((e) => /eslint/.test(e));
+
+                    if (!hasEslint) {
+                        lintChildren?.push("eslint");
+                    }
+                }
+
+                if (this.checkChildrenScripts(pkg, "lint", lintChildren, false, fix)) {
+                    fixed = true;
+                }
             }
-            if (
-                this.checkChildrenScripts(
-                    pkg,
-                    "lint:fix",
-                    lintChildren.map((value) => `${value}:fix`),
-                    false,
-                    fix,
-                )
-            ) {
-                fixed = true;
+
+            // Check "lint:fix"
+            if (lintFixScript) {
+                const lintFixMatch = lintFixScript?.match(lintRegex);
+                const lintFixChildren = lintFixMatch?.map((e) => e.replace(/npm run |:fix/g, ""));
+
+                if (lintFixChildren !== undefined) {
+                    const hasEslint = lintFixChildren.some((e) => /eslint/.test(e));
+
+                    if (!hasEslint) {
+                        lintFixChildren?.push("eslint");
+                    }
+                }
+
+                if (
+                    this.checkChildrenScripts(
+                        pkg,
+                        "lint:fix",
+                        lintFixChildren?.map((e) => `${e}:fix`),
+                        false,
+                        fix,
+                    )
+                ) {
+                    fixed = true;
+                }
             }
+
```

Comment:
We appear to still be doing eslint-related checks below here. Is that expected? Or should they be removed?

Reply:
I left the `eslint` and `eslint:fix` checkers below since they contain different scripts than `lint` and `lint:fix`.